### PR TITLE
Added support for "sow" parameter (Split On Whitespace)

### PR DESCRIPTION
**Pull request by thomascorthals**
Using &sow=false is possible since Solr 6.5. It's needed when you're using multi-word synonyms with SynonymGraphFilter.

### CODE REVIEW

File: src/Component/EdisMax.php [hunk 1]
```diff
@@ -200,4 +200,31 @@ public function getUserFields()
...
+
+    /**
+     * Get SplitOnWhitespace option.
+     *
+     * @return bool|null
+     */
+    public function getSplitOnWhitespace()
+    {
+        return $this->getOption('splitonwhitespace');
+    }
```
Comment (mkalkbrenner) [inline src/Component/EdisMax.php]: According to https://lucidworks.com/2017/04/18/multi-word-synonyms-solr-adds-query-time-support/ the `sow`parameter is available for the `edismax` and the `lucene` query parsers.

Depending on your solrconf.xml, you use the lucene query parser if you don't add the EdisMax component to Solarium\QueryType\Select\Query

Therefore I suggest to move this functionality to Solarium\QueryType\Select\Query itself. I assume that the sow parameter simply doesn't hurt if someone sets it in combination with dismax for example.

File: src/Component/RequestBuilder/EdisMax.php [hunk 2]
```diff
@@ -46,6 +46,7 @@ public function buildComponent($component, $request)
         $request->addParam('bf', $component->getBoostFunctions());
         $request->addParam('boost', $component->getBoostFunctionsMult());
         $request->addParam('uf', $component->getUserFields());
+        $request->addParam('sow', null === ($sow = $component->getSplitOnWhitespace()) ? null : ($sow ? 'true' : 'false'));
```
Comment (mkalkbrenner) [inline src/Component/RequestBuilder/EdisMax.php]: The special handling of null and the string representation of booleans is not required here. We already do that internally in addParam().

File: tests/Component/RequestBuilder/EDisMaxTest.php [hunk 3]
```diff
@@ -31,6 +31,7 @@ public function testBuildComponent()
         $component->setBoostFunctions('functionX(price)');
         $component->setBoostFunctionsMult('functionX(date)');
         $component->setUserFields('title *_s');
+        $component->setSplitOnWhitespace(false);
```
Comment (mkalkbrenner) [inline tests/Component/RequestBuilder/EDisMaxTest.php]: Same as above. The test should be moved to Solarium\Tests\QueryType\Select\RequestBuilderTest

File: src/Component/RequestBuilder/EdisMax.php [hunk 4]
```diff
@@ -46,6 +46,7 @@ public function buildComponent($component, $request)
...
 
         return $request;
     }
```
Comment (mkalkbrenner) [inline src/Component/RequestBuilder/EdisMax.php]: The new parameter should at least be described within docs/queries/select-query/building-a-select-query/building-a-select-query.md

**Comment by thomascorthals** [comment]
I have moved this functionality to Solarium\QueryType\Select\Query and updated the tests.

**Pull request merged**